ArcGISRuntimeSDKQt_CppSamples/Analysis/PerformValveIsolationTrace/PerformValveIsolationTrace.cpp

```diff
@@ -0,0 +1,273 @@
+
+  // get the selected utility category
+  if (categories[m_selectedIndex] != nullptr)
+  {
+    UtilityCategory* selectedCategory = categories[m_selectedIndex];
+    UtilityCategoryComparison* categoryComparison = new UtilityCategoryComparison(selectedCategory, UtilityCategoryComparisonOperator::Exists, this);
+
+    // set the category comparison to the barriers of the configuration's trace filter
+    m_traceConfiguration->filter()->setBarriers(categoryComparison);
+
+    // set whether to include isolated features
+    m_traceConfiguration->setIncludeIsolatedFeatures(m_isolateFeatures);
+
+    // build parameters for the isolation trace
+    UtilityTraceParameters* traceParameters = new UtilityTraceParameters(UtilityTraceType::Isolation, QList<UtilityElement*> {m_startingLocation}, this);
+    traceParameters->setTraceConfiguration(m_traceConfiguration);
+
+    m_utilityNetwork->trace(traceParameters);
+  }
+}
+
+void PerformValveIsolationTrace::connectSignals()
+{
+  connect(m_utilityNetwork, &UtilityNetwork::doneLoading, this, [this](const Error& error)
+  {
+    if (!error.isEmpty())
+    {
+      qDebug() << error.message() << error.additionalMessage();
+      return;
+    }
+
+    if (m_utilityNetwork->loadStatus() != LoadStatus::Loaded)
+      return;
+
+    // get a trace configuration from a tier
+    UtilityNetworkDefinition* networkDefinition = m_utilityNetwork->definition();
+    UtilityDomainNetwork* domainNetwork = networkDefinition->domainNetwork("Pipeline");
+    UtilityTier* tier = domainNetwork->tier("Pipe Distribution System");
+    m_traceConfiguration = tier->traceConfiguration();
+
+    // create a trace filter
+    m_traceConfiguration->setFilter(new UtilityTraceFilter(this));
+
+    // get a default starting location
+    UtilityNetworkSource* networkSource = networkDefinition->networkSource("Gas Device");
+    UtilityAssetGroup* assetGroup = networkSource->assetGroup("Meter");
+    UtilityAssetType* assetType = assetGroup->assetType("Customer");
+    m_startingLocation = m_utilityNetwork->createElementWithAssetType(assetType, QUuid("98A06E95-70BE-43E7-91B7-E34C9D3CB9FF"), nullptr, this);
+
+    // display starting location
+    m_utilityNetwork->featuresForElements(QList<UtilityElement*> {m_startingLocation});
+
+    // populate the combo box choices
+    m_categoriesList = categoriesList();
+    emit categoriesListChanged();
+  });
+
+  connect(m_utilityNetwork, &UtilityNetwork::traceCompleted, this, [this](QUuid)
+  {
+    m_uiEnabled = true;
+    emit uiEnabledChanged();
+
+    UtilityTraceResultListModel* utilityTraceResultList = m_utilityNetwork->traceResult();
+
+    if (utilityTraceResultList->isEmpty())
+    {
+      qWarning() << "No results found.";
+      return;
+    }
+
+    if (UtilityElementTraceResult* utilityElementTraceResult = dynamic_cast<UtilityElementTraceResult*>(utilityTraceResultList->at(0)))
+    {
+      if (utilityElementTraceResult->elements(this).empty())
```

Comment:
this looks like it is creating a bunch of `elements` which use `this` as the parent - but since we don't store them in any way they are effectively a leak. Not a big deal if this can only be run once